File: core/src/main/java/org/dreamexposure/discal/core/enums/event/EventFrequency.java
package org.dreamexposure.discal.core.enums.event;

/**
 * Created by Nova Fox on 11/10/17.
 * Website: www.cloudcraftgaming.com
 * For Project: DisCal-Discord-Bot
 */
public enum EventFrequency {
    HOURLY, DAILY, WEEKLY, MONTHLY, YEARLY;

    /**
     * Checks if the value is a valid enum value.
     *
     * @param value The value to check.
     * @return {@code true} if valid, else {@code false}.
     */
    public static boolean isValid(final String value) {
        return "DAILY".equalsIgnoreCase(value) || "WEEKLY".equalsIgnoreCase(value) || "MONTHLY".equalsIgnoreCase(value)
            || "YEARLY".equalsIgnoreCase(value);
    }

    /**
     * Gets the enum value for the specified string value.
     *
     * @param value The value to get from.
     * @return The enum value.
     */
    public static EventFrequency fromValue(final String value) {
        switch (value.toUpperCase()) {
            case "WEEKLY":
                return WEEKLY;
            case "MONTHLY":
                return MONTHLY;
            case "YEARLY":
                return YEARLY;
            default:
                return DAILY;
        }
    }

    public String getName() {
        return this.name();
    }
}

File: core/src/main/kotlin/org/dreamexposure/discal/core/enums/event/EventFrequency.kt
package org.dreamexposure.discal.core.enums.event

import java.util.*

enum class EventFrequency {
    DAILY, WEEKLY, MONTHLY, YEARLY;

    companion object {
        fun isValid(value: String): Boolean {
            return value.equals("DAILY", true)
                    || value.equals("WEEKLY", true)
                    || value.equals("MONTHLY", true)
                    || value.equals("YEARLY", true)
        }

        fun fromValue(value: String): EventFrequency {
            return when (value.uppercase(Locale.getDefault())) {
                "WEEKLY" -> WEEKLY
                "MONTHLY" -> MONTHLY
                "YEARLY" -> YEARLY
                else -> DAILY
            }
        }
    }
}
